data_model/src/trigger/mod.rs

```diff
@@ -20,22 +21,12 @@ pub mod set;
 
 /// Type which is used for registering a `Trigger`.
 #[derive(
-    Debug,
-    Display,
-    Clone,
-    PartialEq,
-    Eq,
-    Decode,
-    Encode,
-    Deserialize,
-    Serialize,
-    FfiType,
-    IntoSchema,
+    Debug, Display, Clone, Decode, Encode, Deserialize, Serialize, FfiType, IdOrdEqHash, IntoSchema,
 )]
 #[display(fmt = "@@{id}")]
 pub struct Trigger<F: Filter> {
     /// [`Id`] of the [`Trigger`].
-    pub id: <Trigger<FilterBox> as Identifiable>::Id,
+    pub id: Id,
```

Comment:
@Arjentix are we ok with this change?

Reply:
my question here is whether it's ok to derive `IdEqOrdHash` for `Trigger<F>` whereas previously it was constrained to `Trigger<FilterBox>`

Reply:
Hm..., interesting question

Reply:
Well, it was implemented only on `Trigger<FilterBox>` so that noone accidentally store other triggers as `Identifiable`. From the Iroha-user point of view only `Trigger<FilterBox>` can be identifiable.

Reply:
So in general this is not a big deal, I believe.

Maybe you can add `bound` attr which will allow to add *derive constrains*, just like serde [does](accidentally).
But that could be an extra work for just one little thing

Reply:
So you can keep it as is, but I like more the approach with `bound` attribute

Reply:
I believe your concern is covered by the `Registered` implementation for `Trigger` and that the `Trigger` regardless of the type is uniquely identifiable

Reply:
From my point of view, deriving a trait should not have unintended side-effects. The very fact that we're having this discussion suggests to me that the delineation should be a little more explicit in general. 

I recommend adding `bound` in a separate PR. Leaving things as is for the time being.